Assets/Scripts/Animation/PopupInventoryItem.cs

```diff
@@ -1,105 +1,73 @@
     }
 
    private IEnumerator GoUp() {
-        float currHeight = gameObjectImage.transform.position.y;
-        while (currHeight < maxHeight) {
-            gameObjectImage.transform.Translate(0.0f, bounceMagnitude * Time.deltaTime, 0.0f);
-            currHeight = gameObjectImage.transform.position.y;
+        while (transform.position.y < originalPos.y + bounceHeight) {
+            transform.Translate(0.0f, bounceSpeed * Time.deltaTime, 0.0f);
             yield return null;
         }
      }
 
     private IEnumerator GoDown() {
-        float currHeight = gameObjectImage.transform.position.y;
-        while (currHeight > originalPos.y) {
-            gameObjectImage.transform.Translate(0.0f, -bounceMagnitude * Time.deltaTime, 0.0f);
-            currHeight = gameObjectImage.transform.position.y;
+        while (transform.position.y > originalPos.y) {
+            transform.Translate(0.0f, -bounceSpeed * Time.deltaTime, 0.0f);
             yield return null;
         }
     }
+
     /// <summary>
     /// Animate the item sprite in an up and down motion
     /// </summary>
-    /// <returns></returns>
     private IEnumerator AnimateSprite() {
-        itemRendered = false;
-        for (int animLoop = 0; animLoop < 3; animLoop++) {
+        for (int animLoop = 0; animLoop < bounceRepetitions; animLoop++) {
             yield return StartCoroutine(GoUp());
             yield return StartCoroutine(GoDown());
-            yield return new WaitForSeconds(0.1f);
-        }
-        SetImageSprite(null);
-        SetImageActive(false);
-        ResetImagePosition();
-    }
-    /// <summary>
-    /// stops animation of the item 
-    /// </summary>
-    /// <param name="active">determines whether to activate stop animation</param>
-    /// <returns></returns>
-    public bool StopAnimation(bool active) {
-        if (active) {
-            StopAllCoroutines();
-            return true;
+            yield return null;
         }
-        return false;
-    }
-    /// <summary>
-    /// assigns sprite to popup item sprite
-    /// </summary>
-    /// <param name="sprite"></param>
-    public static void SetImageSprite(Sprite sprite) {
-        gameObjectImage.sprite = sprite;
+        ResetImage();
     }
 
     /// <summary>
-    /// Set whether sprite is visible in-game or not
+    /// Stops animation of the item 
     /// </summary>
-    /// <param name="active"></param>
-    public static void SetImageActive(bool active) {
-        Color currcolor = gameObjectImage.color;
-        currcolor.a = (active) ? 1 : 0;
-        gameObjectImage.color = currcolor;
-    }
-    /// <summary>
-    /// after each animation, reset the popup item position to its original position
-    /// </summary>
-    public static void ResetImagePosition() {
-        gameObjectImage.transform.position = originalPos;
+    public void StopAnimation() {
+        StopAllCoroutines();
+        ResetImage();
     }
 
+    private void ResetImage() {
```

Comment:
I consolidated these since they are often called together, and it isn't really necessary to have a separate function for a single line of code.